### ISSUE TITLE
Trying to test with v0.6.2, getting a cargo error

### ISSUE THREAD

**Issue by winding-lines** [serde-rs/xml#13]
Something may have gone wrong with the latest publish. I used `cargo update` to I upgraded my library (graphml), it got version 0.6.2 of `serde_xml` but when I am trying to run the tests I am getting the following error:

```
$ multirust run nightly cargo test --verbose
multirust: checking metadata version
multirust: got metadata version 2
unable to get packages from source

Caused by:
  failed to unpack package `serde_xml v0.6.2`

Caused by:
  No such file or directory (os error 2)
```

Switching to a git url works.

**Comment by winding-lines** [comment]
Yay with your latest changes all my tests are passing, thanks a lot! 

For now using a git url for `serde-xml` blocks the publish, but at least I pushed the actual code.

```
$ cargo publish
    Updating registry `https://github.com/rust-lang/crates.io-index`
all dependencies must come from the same source.
dependency `serde_xml` comes from https://github.com/serde-rs/xml.git instead
```

After you retry the publish I will try one more time. This way we can get you one reverse dependence ;-)
[https://crates.io/crates/serde_xml/reverse_dependencies](https://crates.io/crates/serde_xml/reverse_dependencies)

**Comment by oli-obk** [comment]
I checked out `graphml` and changed the `serde_xml` dependency from the git to "0.6.2" and it works fine. Maybe you had a broken download? have you tried running `cargo update` again? If that fails, try clearing your cargo's cache. It might have gotten a broken package during downloading and not done a checksum (not sure if it ever does a checksum)

**Comment by matthewkmayer** [comment]
I'm seeing this as well.  Just added `serde_xml = "0.6.2"` to my Cargo.toml file.  Ran:

`rm -fr ~/.cargo/registry/*`

`cargo update`

``` bash
$ cargo build --verbose
 Downloading kernel32-sys v0.2.1
 Downloading term v0.2.14
 Downloading hyper v0.7.2
 Downloading bitflags v0.4.0
 Downloading openssl-sys-extras v0.7.6
 Downloading bitflags v0.3.3
 Downloading log v0.3.5
 Downloading rand v0.3.14
 Downloading rust-crypto v0.2.34
 Downloading httparse v1.1.1
 Downloading rustc-serialize v0.3.18
 Downloading rustc_version v0.1.7
 Downloading user32-sys v0.1.2
 Downloading libressl-pnacl-sys v2.1.6
 Downloading openssl-sys v0.7.6
 Downloading pkg-config v0.3.7
 Downloading chrono v0.2.19
 Downloading syntex_syntax v0.29.0
 Downloading unicase v1.3.0
 Downloading mime v0.1.3
 Downloading gdi32-sys v0.1.1
 Downloading quasi v0.7.0
 Downloading winapi v0.2.5
 Downloading aster v0.13.1
 Downloading quasi_codegen v0.7.0
 Downloading serde_json v0.6.0
 Downloading openssl v0.7.6
 Downloading pnacl-build-helper v1.4.10
 Downloading utf8-ranges v0.1.3
 Downloading serde_xml v0.6.2
unable to get packages from source

Caused by:
  failed to unpack package `serde_xml v0.6.2`

Caused by:
  No such file or directory (os error 2)
```

Branch with the error: https://github.com/rusoto/rusoto/tree/serde-xml-error .

**Comment by oli-obk** [comment]
I cannot reproduce this. Maybe your cargo cache is corrupted.

```
$ cargo build
    Updating registry `https://github.com/rust-lang/crates.io-index`
 Downloading Inflector v0.1.6
 Downloading solicit v0.4.4
 Downloading pkg-config v0.3.7
 Downloading chrono v0.2.19
 Downloading openssl v0.7.6
 Downloading httparse v1.1.1
 Downloading uuid v0.1.18
 Downloading syntex_syntax v0.29.0
 Downloading hyper v0.7.2
 Downloading hpack v0.2.0
 Downloading unicode-bidi v0.2.3
 Downloading pnacl-build-helper v1.4.10
 Downloading openssl-sys v0.7.6
 Downloading semver v0.1.20
 Downloading openssl-sys-extras v0.7.6
 Downloading url v0.2.38
 Downloading gcc v0.3.24
 Downloading Inflector v0.2.0
 Downloading botocore_parser v0.1.1
 Downloading syntex v0.29.0
 Downloading libressl-pnacl-sys v2.1.6
 Downloading typeable v0.1.2
 Downloading unicase v1.3.0
 Downloading url v0.5.5
 Downloading language-tags v0.2.2
 Downloading num_cpus v0.2.11
 Downloading tempdir v0.3.4
 Downloading rustc_version v0.1.7
 Downloading matches v0.1.2
 Downloading rust-crypto v0.2.34
 Downloading cookie v0.2.2
 Downloading regex v0.1.54
 Downloading mime v0.1.3
 Downloading regex-syntax v0.2.5
 Downloading traitobject v0.0.1
   Compiling winapi v0.2.5
   Compiling bitflags v0.4.0
   Compiling lazy_static v0.1.15
   Compiling typeable v0.1.2
   Compiling winapi-build v0.1.1
   Compiling utf8-ranges v0.1.3
   Compiling unicode-xid v0.0.3
   Compiling kernel32-sys v0.2.1
   Compiling rustc-serialize v0.3.18
   Compiling regex-syntax v0.2.5
   Compiling matches v0.1.2
   Compiling unicode-bidi v0.2.3
   Compiling term v0.2.14
   Compiling unicode-normalization v0.1.2
   Compiling bitflags v0.3.3
   Compiling traitobject v0.0.1
   Compiling xml-rs v0.1.26
   Compiling libc v0.1.12
.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/libc-0.1.12/rust/src/liblibc/lib.rs:81:21: 81:39 warning: lint raw_pointer_derive has been removed: using derive with raw pointers is ok
.multirust/toolchains/nightly/cargo/registry/src/github.com-88ac128001ac3a9a/libc-0.1.12/rust/src/liblibc/lib.rs:81 #![allow(bad_style, raw_pointer_derive)]
                                                                                                                                                        ^~~~~~~~~~~~~~~~~~
   Compiling pkg-config v0.3.7
   Compiling httparse v1.1.1
   Compiling gcc v0.3.24
   Compiling openssl-sys v0.7.6
   Compiling language-tags v0.2.2
   Compiling openssl-sys-extras v0.7.6
   Compiling openssl v0.7.6
   Compiling rust-crypto v0.2.34
   Compiling libc v0.2.7
   Compiling semver v0.1.20
   Compiling num_cpus v0.2.11
   Compiling time v0.1.34
   Compiling log v0.3.5
   Compiling rustc_version v0.1.7
   Compiling syntex_syntax v0.29.0
   Compiling unicase v1.3.0
   Compiling memchr v0.1.10
   Compiling rand v0.3.14
   Compiling aho-corasick v0.5.1
   Compiling regex v0.1.54
   Compiling uuid v0.1.18
   Compiling url v0.5.5
   Compiling url v0.2.38
   Compiling cookie v0.2.2
   Compiling hpack v0.2.0
   Compiling Inflector v0.2.0
   Compiling Inflector v0.1.6
   Compiling solicit v0.4.4
   Compiling num v0.1.31
   Compiling chrono v0.2.19
   Compiling serde v0.6.15
   Compiling mime v0.1.3
   Compiling serde_json v0.6.0
   Compiling hyper v0.7.2
   Compiling quasi v0.7.0
   Compiling aster v0.13.1
   Compiling syntex v0.29.0
   Compiling quasi_codegen v0.7.0
   Compiling serde_codegen v0.6.14
   Compiling botocore_parser v0.1.1
   Compiling rusoto v0.12.0 (rusoto)
failed to run custom build command for `rusoto v0.12.0 (rusoto)`
Process didn't exit successfully: `rusoto/target/debug/build/rusoto-2e693f164d7ea6bf/build-script-build` (exit code: 101)
--- stderr
thread '<main>' panicked at 'called `Result::unwrap()` on an `Err` value: Error { repr: Os { code: 2, message: "No such file or directory" } }', ../src/libcore/result.rs:746
note: Run with `RUST_BACKTRACE=1` for a backtrace.

```

**Comment by matthewkmayer** [comment]
I thought the `rm` command I ran earlier cleaned the cache, guess I was mistaken.  Our [Travis build](https://travis-ci.org/rusoto/rusoto/jobs/111386472) works fine on the branch.  I'll go back to researching what's up with my cargo cache, thanks!

**Comment by oli-obk** [comment]
I'm seeing the same issue right now if I use stable... this is weird. Nightly works, stable doesn't. How can that be?

**Comment by oli-obk** [comment]
I pushed `0.6.3` with stable (and whatever cargo version multirust bundles with stable). Now it works for me... If that doesn't work, also delete Cargo.lock instead of using `cargo update`

**Comment by oli-obk** [comment]
@matthewkmayer @winding-lines did this get fixed for you by `0.6.3`?

**Comment by matthewkmayer** [comment]
Haven't seen this happen again, I think it's all good.  Thanks!
